lxd/storage/backend_lxd.go

```diff
@@ -5951,6 +5951,32 @@ func (b *lxdBackend) detectChangedConfig(curConfig, newConfig map[string]string)
 	return changedConfig, userOnly
 }
 
+func allowRemoveSecurityShared(s *state.State, projectName string, volume *api.StorageVolume) error {
+	err := VolumeUsedByProfileDevices(s, volume.Pool, projectName, volume, func(profileID int64, profile api.Profile, project api.Project, usedByDevices []string) error {
+		return fmt.Errorf("Cannot disable security.shared on custom storage block volume as it is attached to profile(s)")
+	})
+	if err != nil {
+		return err
+	}
+
+	var usedByInstanceDevices []string
+
+	err = VolumeUsedByInstanceDevices(s, volume.Pool, projectName, volume, true, func(inst db.InstanceArgs, project api.Project, usedByDevices []string) error {
+		usedByInstanceDevices = append(usedByInstanceDevices, inst.Name)
```

Comment:
this caught my eye. Why do we need this usedByInstanceDevices slice, when the very act of this function being called means it is in use and thus we can just return an error?

Reply:
Agreed, the slice is completely unnecessary. But, removing `security.shared` is perfectly valid if there are 0 or 1 instances using the volume.